### ISSUE TITLE
.e Nervus file reading doesn't work in Octave

### ISSUE THREAD

**Issue by programmernoob1** [fieldtrip/fieldtrip#1766]
**DESCRIPTION**

There are some functions like datetime used in read_nervus_header.m in filoio/private that is missing in Octave. Thus, it is failing.
This worked perfectly fine in Matlab enviroment. The changes work fine since the last commit for the file - https://github.com/fieldtrip/fieldtrip/commit/2834b5ac308b3423d99b3e1bb0aeacbd560d111a#diff-22cf50ab771575b65dc6737080744f6ea45d020b56327797a89370d3bc35ff4d .  

**TO REPRODUCE**

1. Open octave-cli
2. create cfg for .e file.

```
nicoletfile = 'x.e'
        cfg            = [];
        cfg.dataset    = nicoletfile;
        cfg.continuous = 'yes';
        cfg.channel    = 'all';
        
```
3. Run ft_preprocessing
```
data           = ft_preprocessing(cfg);
```

**EXPECTED BEHAVIOUR**

ft_preprocessing worked perfectly fine in Matlab environment. But, it is failing for Octave.
ft_processing should have loaded the eeg data into the variable.

**Environment and versions (please complete the following information):**
 - Operating System: MacOs Big Sur 11.2.3
 - Octave - 6.2.0
 -  FieldTrip version -master

PS: This is my first issue on Github. Please mention if something else is required to work on this issue.

**Comment by robertoostenveld** [comment]
Dear noob, 

We are not actively supporting Octave, but see https://www.fieldtriptoolbox.org/faq/can_i_use_octave_instead_of_matlab. But if - with little effort - we can get slightly more code running on Octave, we will try to do that. Note however that I do not have Octave installed so cannot test anything myself.

What I don't understand from your issue is whether reading the Nicolet `.e` file in Octave worked in the past.Does a fieldtrip version prior to 27 May 2019 (i.e. prior to https://github.com/fieldtrip/fieldtrip/commit/2834b5ac308b3423d99b3e1bb0aeacbd560d111a#diff-22cf50ab771575b65dc6737080744f6ea45d020b56327797a89370d3bc35ff4d) work for you? It would probably help if you would share the precise error message that you get on screen.

**Comment by robertoostenveld** [comment]
if that is the case, you (i.e. @programmernoob1) might want to check https://github.com/fieldtrip/fieldtrip/commits/master/fileio/private/read_nervus_header.m and https://github.com/fieldtrip/fieldtrip/commits/master/fileio/private/read_nervus_data.m, and the commit changing `fopen` by `fopen_or_error` would not be related.

**Comment by programmernoob1** [comment]
Hi @robertoostenveld, 
       I used the read_nervus_header code from the previous commit in that file commit history. That helped me run the read_nervus_header function. But, I came across another error in ft_preprocessing in line no. 437(https://github.com/fieldtrip/fieldtrip/blob/6fb710a21ebf99dfec149c367bc8920bb56ee7c4/ft_preprocessing.m#L437) , where istable function is used. The function "istable" is not available in Octave.
       
       I tried to pull the tag - 20191213  and run ft_preprocessing before "istable" was introduced, but that gave me another error  -  startsWith is not defined. startsWith.m was added to the repo just 2 months ago.
       
       
In short, which tag would you recommend for me to use for Octave to deal with Nervus files? Thanks.

**Comment by robertoostenveld** [comment]
We are not supporting nor testing FieldTrip with Octave, so I cannot give you recommendations on which version of FieldTrip might work best with which version of Octave. I suggest you implement a function `fieldtrip/compat/octave/istable.m` that behaves like the MATLAB counterpart. 

You may want to try https://github.com/apjanke/octave-tablicious to provide the missing functionality in Octave. Its author @apjanke has also contributed to FieldTrip, so he might want to chime in here directly.

**Comment by programmernoob1** [comment]
Okay, got it. Thank you for taking your time to reply. I shall try this out.
